frontend/src/components/maps/leaflet/InfoSlideOut/ProjectNumberLink.tsx

```diff
@@ -0,0 +1,88 @@
+import Claims from 'constants/claims';
+import { Workflows } from 'constants/workflows';
+import { fetchProject, IProject } from 'features/projects/common';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+interface IProjectNumberLinkProps {
+  /** The project number to analyze */
+  projectNumber: string;
+  /** The agency ID of the property being viewed */
+  agencyId: number | '';
+  /** To determine whether the project is private to the user or not */
+  setPrivateProject: Function;
+  /** The result of setPrivateProject used in both the parent and child component */
+  privateProject: boolean;
+}
+
+/**
+ * This component is used to display project numbers and appropriate links if the user has permission.
+ * A user who belongs to the same agency and a project that has not been submitted will grant user access to a link that goes directly to the status.
+ * An admin will have access to a link that goes directly to the status.
+ * If a user belongs to the agency but the project has been submitted it will redirect you to the summary page.
+ * If a user is outside the agency they will not contain a link.
+ * @param {string} projectNumber The project number to analyze
+ * @param {number} agencyId The agency ID of the property being viewed
+ * @param {Function} setPrivateProject Will set the parent state to determine whether the project can be viewed by user
+ * @param {boolean} privateProject The resulting boolean of setPrivateProject that is used in the logic in determining whether the number will be a link
+ */
+export const ProjectNumberLink: React.FC<IProjectNumberLinkProps> = ({
+  projectNumber,
+  agencyId,
+  setPrivateProject,
+  privateProject,
+}) => {
+  const dispatch = useDispatch();
+  const keycloak = useKeycloakWrapper();
+  /** Stores the various states of the project route (summary page, project status route) */
+  const [projectRoute, setProjectRoute] = useState<string>('');
+  /** State used to determine whether further project info has been fetcehd already */
+  const [loaded, setLoaded] = useState<boolean>(false);
```

Comment:
Don't really like doing this, but it was making me add keycloak to the dependency array and that would cause an infinite loop so created a state variable that will execute and determine whether the component still needs to get the project status route

Reply:
keycloak "shouldn't" cause infinite loop.  But once again the dep array is killing us.  I suspect the deepcompare would solve it, but Devin will know more.

Reply:
I tried `useDeepCompareEffect` as well.. I must be missing something somewhere. Just not sure of what..